### PR TITLE
Build microovn from last compatible version

### PR DESCRIPTION
**Pull request by masnax**
Until MicroCeph updates its microcluster dependency to `v2`, we're stuck using microcluster `v1` in MicroCloud which is incompatible with `v2` because the core endpoint prefixes have changed.

To work around this in the mean time, this adds a block to the github workflows to build MicroOVN from the last compatible commit. The built snap is then supplied to the test suite.

This is actually easier to clean up than trying to fudge both `v1` and `v2` microcluster into MicroCloud, which would require some larger refactors. Once we're ready to move to `v2`, we can just remove the block from the workflow.

### CODE REVIEW

File: test/main.sh [hunk 1]
```diff
@@ -132,9 +132,12 @@ export MICROCEPH_SNAP_CHANNEL
 MICROCLOUD_SNAP_CHANNEL="${MICROCLOUD_SNAP_CHANNEL:-latest/edge}"
 export MICROCLOUD_SNAP_CHANNEL
 
-MICROOVN_SNAP_CHANNEL="${MICROOVN_SNAP_CHANNEL:-latest/edge}"
+MICROOVN_SNAP_CHANNEL="${MICROOVN_SNAP_CHANNEL:-22.03/stable}"
```
Comment (roosterfish) [inline test/main.sh:135]: Do we still need this env variable as MicroOVN is now built from source?

File: test/suites/basic.sh [hunk 2]
```diff
@@ -826,8 +826,8 @@ test_service_mismatch() {
 
   # Install microceph and microovn on the first machine only.
   for m in micro02 micro03 ; do
-    lxc exec "${m}" -- snap remove microceph --purge
-    lxc exec "${m}" -- snap remove microovn --purge
+    lxc exec "${m}" -- snap disable microceph
```
Comment (roosterfish) [inline test/suites/basic.sh:829]: Can it cause some leftovers if we just disable/enable the snaps? 
Why not running ` snap install --dangerous "/root/microovn.snap"` here?

**Review by roosterfish — CHANGES_REQUESTED** [review]
I like the idea so we can unblock the test suite for the time being. Just two small comments.

File: test/main.sh (hunk 1 above)
Comment (masnax) [reply test/main.sh:135]: Yes, the recommended way to install a built snap is to install the default snap first and then use `--dangerous`. This is to preserve any snap interfaces that won't be set up normally with just `--dangerous` or `--devmode`.

File: test/suites/basic.sh (hunk 2 above)
Comment (masnax) [reply test/suites/basic.sh:829]: It's overall quicker to simply disable/re-enable the snaps, which is what we do all over the test suite. This one spot was an anomaly.

File: test/main.sh (hunk 1 above)
Comment (roosterfish) [reply test/main.sh:135]: I see thanks!

**Review by roosterfish — COMMENTED** [review]
Might the pipeline errors be related to https://github.com/golangci/golangci-lint/issues/4909?

**Comment by masnax** [comment]
> Might the pipeline errors be related to [golangci/golangci-lint#4909](https://github.com/golangci/golangci-lint/issues/4909)?

Sigh, Added a line to comment out golangci-lint in `snapcraft.yaml`

**Review by roosterfish — APPROVED** [review]

**Pull request merged**